### PR TITLE
feature/mainari

### PR DESCRIPTION
**Pull request by Zackyist**
### A new module for a new Minecraftian age

Mainari is a module supporting Minecraft related queries. This pull request:

- adds the /minecraft command to get basic server status
- **requires** the appropriate Minecraft related env variables in env.cfg (see env-example.cfg)
- adds support for command line arguments
- adds the --verbose command line argument to provide easy debug level logging
- adds the Visual Studio Code project settings file to .gitignore
- adds a mention about the Minecraft env variables to the readme

A few notes:

- uses the https://api.mcsrvstat.us API to get the server data
- the /minecraft command has a cooldown of 60 seconds to prevent online API abuse (and prevent the bot from getting banned from the API)
- ingame ops get a @ before their name in the player list when online. This does **not** result in a faulty Telegram mention due to Markdown formatting
- server admins will all get Telegram mentions if the server is down. Better not ruin that uptime, guys
- Mainari strips any leading and trailing whitespace from the admin lists so you can have spaces between the names, just don't forget your commas!

### CODE REVIEW

File: mainari.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    def parseNicks(self, nicks):
+        nick_list = nicks.split(',')
+        nick_list = [nick.strip() for nick in nick_list]
+        return nick_list
+
+
+    def parseServerData(self, data):
+        # Data gathered if the server is offline
+        try:
+            server_is_offline = data['offline']
```
Comment (Aivoapina) [inline mainari.py]: You can check if key is in dict with `if 'offline' in data:` instead of try except.

File: mainari.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        except:
+            server_is_offline = False
+            server_status_msg = 'Online'
+            motd = data['motd']['raw'][0]
+            players = data['players']['online']
+            players_max = data['players']['max']
+            version = data['version']
+
+            try:
+                player_list = data['players']['list']
```
Comment (Aivoapina) [inline mainari.py]: Nested try except blocks aren't best practice. Use `key in (not) dict` instead

File: mainari.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+            except:
+                player_list = []
+    
+            try:
+                plugins = data['plugins']['raw']
```
Comment (Aivoapina) [inline mainari.py]: Same here

File: mainari.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+            except:
+                plugins = []
+        
+        # A precaution against errors if the server is using an IP instead
+        try:
+            hostname = data['hostname']
```
Comment (Aivoapina) [inline mainari.py]: And here

**Comment by Zackyist** [comment]
Finally, after almost a year, I took the time to make the necessary changes (as well as some additional ones).

Try blocks are now if blocks, a newer server status API version is in use, support for voluntarily showing the IP address instead of the hostname now exists, and the /minecraft command cooldown was bumped up from 45 seconds to 60 seconds. Also merged the latest changes from master.

**Pull request merged**